content/Phoenix for Rails developers - Bruno Vezoli.md

```diff
@@ -0,0 +1,347 @@
+
+This way of managing the data layer and the business logic opens up a world of oportunities in my opinion. Assume for example that we have a process for publishing in our blo, so all articles must be reviewed after they are changed and before they are published. We would probably do some validations in our model to only insert it if it's approved, same would go in our context here. But we also like to allow admins to bypass these validations, a common case, and publish whenever they want. So in Rails we would probably change our code to reflect this new case while in Phoenix we might wish to create a new context all toghether.
+
+As with any choice you always have pros and cons, but I really like the way context allow you to avoid polluting a model with different responsibilities. If you've been working on Rails long enough you might notice they have a similar purpose as service objects which the community has embraced and which are a part of our daily development cycle.
+
+### Helpers and Views
+
+Last but not least is how we present the data. You probably noticed in the file structure that there where two directories called `views/` and `templates/` on the Phoenix project structure, if you didn't go and take a closer view. You might be wondering what's the difference between them, well the templates are just that, text files with embedded code (just like erb we have eex) which you use to respond to requests. Views on the other hand resemble to helpers but with one noticeable difference: they are not global by default.
+
+Let's take a look at how they differ with an example; we would like to show the read time in our articles. Since we don't track this value on our DB and it's just for presentation you'd normally want to put this on a helper, like this:
+
+```ruby
+module ArticlesHelper
+  WORDS_PER_MINUTE = 200
+
+  def read_time(article)
+    words_count = article.text.split.size
+    read_time_in_minutes = words_count / WORDS_PER_MINUTE.to_f
+    “#{read_time_in_minutes.ceil} minutes"
+  end
+end
+```
+
+Easy peasy, now we just have to call this on our `show.html.erb` and we are done. Not so fast, the problem I see with this is that it gets included on every view. Sure for a small app like this and with a name like so specific there's no problem, but when it starts to grow and you have helpers with names like `format_title` and `format_title_for_user` you start having troubles. Phoenix's helpers (or views as they are called in the framework) are very similar:
```

Comment:
`Sure for a small app like this and with a name like so specific there's no problem,` I would remove the `Sure` and `like` 
Also `start having trouble`